bibernate-core/src/main/java/com/bobocode/petros/bibernate/session/jdbc/DefaultJdbcQueryManager.java

```diff
@@ -1,46 +1,162 @@
 package com.bobocode.petros.bibernate.session.jdbc;
 
+import com.bobocode.petros.bibernate.exceptions.JdbcOperationException;
+import com.bobocode.petros.bibernate.session.query.QueryBuilder;
+import com.bobocode.petros.bibernate.session.query.QueryResult;
 import com.bobocode.petros.bibernate.session.query.condition.Restriction;
+import com.bobocode.petros.bibernate.session.query.enums.QueryType;
 import com.bobocode.petros.bibernate.transaction.Transaction;
+import com.bobocode.petros.bibernate.utils.EntityUtils;
+import com.bobocode.petros.bibernate.utils.StatementUtils;
+import lombok.extern.slf4j.Slf4j;
 
 import javax.sql.DataSource;
+import java.lang.reflect.Field;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
 
+import static java.sql.PreparedStatement.RETURN_GENERATED_KEYS;
+
+/**
+ * Implementation of {@link JdbcQueryManager}.
+ */
+@Slf4j
 public class DefaultJdbcQueryManager implements JdbcQueryManager {
 
-    private DataSource dataSource;
+    private static final String GENERATED_KEY = "GENERATED_KEY";
+
+    private final DataSource dataSource;
 
-    public DefaultJdbcQueryManager(DataSource dataSource) {
-        Objects.requireNonNull(dataSource, "Parameter [dataSource] must be provided!");
+    public DefaultJdbcQueryManager(final DataSource dataSource) {
+        Objects.requireNonNull(dataSource, "Parameter [dataSource] must not be null!");
         this.dataSource = dataSource;
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * @param entity entity that is supposed to be persisted
+     * @param <T>    generic type
+     * @return persisted entity with id
+     */
     @Override
-    public <T> T persist(T entity) {
-        return null;
+    public <T> T persist(final T entity) {
+        Objects.requireNonNull(entity, "Parameter [entity] must not be null!");
+        final String sql = QueryBuilder.buildQuery(entity.getClass(), QueryType.INSERT, Collections.emptyList());
+        try (final Connection connection = this.dataSource.getConnection();
+             final PreparedStatement statement = StatementUtils.prepareInsertStatement(connection.prepareStatement(sql, RETURN_GENERATED_KEYS), entity)) {
```

Comment:
As a proposal for refactoring - update StatementUtils method signature to receive connection and entity
`StatementUtils.createInsertStatement(connection,entity)`
so StatementUtils will be responsible for PreparedStatement creation and building SQL queries. That API will make StatementUtils fully responsible for Statement creation

Reply:
I've introduced another solution) please take a look.